gretl/src/main/java/ch/so/agi/gretl/api/Connector.java

```diff
@@ -4,21 +4,23 @@
 
 import ch.so.agi.gretl.logging.GretlLogger;
 import ch.so.agi.gretl.logging.LogEnvironment;
+import kotlin.jvm.Transient;
```

Comment:
Ist das eine von Gradle bereitgestellte Abhängigkeit? Was macht das genau? Und warum nur in dieser Klasse?

Reply:
@edigonzales Nice catch. :) Der Import konnte entfernt werden, ich hatte zuerst probiert transient als Annotation zu benutzen. Das geht aber nur in Kotlin. Ich habe vergessen den Import wieder zu entfernen.😅 

Transient ist ein Java Keyword, welches einem Objekt, welches Serializable implementiert (hier Connector.java) sagt, dass das mit transient markierte Property nicht serialisiert werden soll. Das hat grundsätzlich nichts mit dem GretlLogger an sich zu tun, sondern hängt mit dem Connector Objekt zusammen. Wir verwenden z.B. in CsvExport.java ein Connector Objekt als Taskinput. Damit Gradle den Cache korrekt aufbauen kann, um zu sehen, ob sich Objekte verändert haben zwischen zwei Runs, muss alles serialisierbar sein. Gradle wirft einen Fehler, wenn Connector.java das Serializable Interface nicht implementiert. Der GretlLogger ist ansonsten nicht Teil eines Objektes, welches als Taskinput verwendet wird, daher benötigt es nicht an jeder Stelle wo der GretlLogger vorkommt das transient Keyword.

Warum ich die zwei Properties im Connector als transient markiert habe: Beim GretlLogger macht es aus meiner Sicht keinen Sinn diesen zu serialisieren. Der Status des Objektes wird durch den Logger nicht beeinflusst.  Das Connection Objekt (Connection dbConnection) kann ich nicht serialisierbar machen, da es sich um eine Klasse handelt, die von Java zur Verfügung gestellt wird. Spielt hier aber auch keine Rolle, da sich die Connection in Abhängigkeit von dbUri, dbUser und dbPassword verändert und wir Änderungen also schon tracken.

Anmerkung zum Logger: Dieser wird im Connector gar nicht benötigt. Könnte man diesen auch einfach entfernen?